### ISSUE TITLE
Move to Knative Community owned GSuite account

### ISSUE THREAD

**Issue by bsnchan** [knative/community#86]
## Current State

The Knative GSuite tools use the `google.com` domain at the moment. This affects:
* Google Drive
  * Folks outside of Google can't control the ownership of the community drive itself
* Knative Community Calendar
  * Folks outside of Google have read/write access but don't have admin privileges
* Hangouts
  * We still use Hangouts for a lot of WG meetings which require someone from Google to let folks into the meeting
* Google Groups
  * I believe all the knative-* google groups are tied to the `google.com` GSuite org as well

## Proposal

Create a knative.dev GSuite org for the community and shift all the community owned artifacts to this org.

## Steps

* [x] Add new TOC / Steering members to the org
* [x] Get calendar sharing working
* [x] Get team drive working
* [x] Document setup
* [ ] Update documentation

**Comment by bsnchan** [comment]
Hey @rgregg @isdal - this feels like an easy one to tackle. Is there any chance we can get some budget for this?

**Comment by mattmoor** [comment]
tl;dr I'd like to plug for an increase in the priority of this.

I had a quick sync with @dims and having a community Gsuite org seems to be one of the cornerstones of how they are migrating the pieces of the K8s CI/CD/Release infrastructure from Google-only, to community based.

It sounds like they have significant automation for managing the Gsuite org (see [here](https://github.com/kubernetes/k8s.io/blob/master/groups/groups.yaml)), and by the end of our chat I was in a very "shut up and take my money" mood.

That this is being used to address the exact-same problems on the Kubernetes side sets a very strong precedent that we didn't have previously motivating this, and the ways they are using it (which we should follow) make this blocking for other items on this board.

**Comment by rgregg** [comment]
I'm working on setting up a G-Suite account for us to use.

**Comment by rgregg** [comment]
We have a G-Suite account now, using `knative.team`. I believe @evankanderson is working on the migration details for our membership groups and shared drive.

**Comment by evankanderson** [comment]
/assign @evankanderson 

We will be moving Team Drive & Calendar on the weekend of July 4 to attempt to minimize disruption.

**Comment by mattmoor** [comment]
@evankanderson You mentioned sitting down with Tomas last week to work on this, any update?

**Comment by evankanderson** [comment]
Update is:

The following calendar and document should be publicly visible:
https://calendar.google.com/calendar?cid=a25hdGl2ZS50ZWFtXzlxODNiZzA3cXM1YjlycnNscDVqb3I0bDZzQGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20
https://docs.google.com/document/d/1hsAifnnyiOW0kvkArqsY0HIXBDijGOzUC8TuPiUhUxc/edit?usp=sharing

Unfortunately, it appears that I can't copy things from "[Knative Community](https://drive.google.com/drive/u/0/folders/0APnJ_hRs30R2Uk9PVA)" to the new "[Knative Team](https://drive.google.com/drive/u/0/folders/0AM-QGZJ-HUA8Uk9PVA)" drive. (I'm going to see if Grant or Nghia can.)

I'll be out part of next week on vacation, but I think the next steps are:

**This week**
- [ ] Create mailing lists for TOC, Steering, and WG leads to use in ACLs
- [ ] Copy calendar events from the Knative calendar
- [ ] Test moving N documents to the new shared drive
- [ ] Draft email to Knative community about the move of calendar & documents

** Next week **
- [ ] Share new Knative calendar with WG leads with edit permissions
- [ ] Share drive with WG leads, TOC, and knative-dev/knative-users
- [ ] Prepare docs PRs with links to new content
- [ ] Move docs from old Shared Drive to new Drive
- [ ] Send email to knative-dev@ and knative-users@ about the change
- [ ] Clear old calendar with a single continuous "moved to <new calendar link>"
- [ ] Add a "moved to new location" document in old Shared Drive

**Comment by vagababov** [comment]
Evan, will we all be able to move meetings back to meet/hangouts running with gSuite account?
Also does anyone get emails from this domain, or it's purely for drive/calendar?

**Comment by evankanderson** [comment]
With respect to meetings, I'm happy to leave the choice of meeting platform up to the WG leads, assuming we can get meetings recorded.

I think the plan is to have administrative accounts for Steering & TOC, but not to create a large number of accounts in the @knative.team domain. It's been tested that accounts from outside the domain can manage sharing for both calendar and drive, so it shouldn't be necessary to give everyone an account within the domain.

**Comment by evankanderson** [comment]
(We may end up moving some mailing lists to @knative.team as well to enable using the Google Apps APIs to manage the mailing lists, but I expect that to be a separate project if it happens.)

**Comment by abrennan89** [comment]
> With respect to meetings, I'm happy to leave the choice of meeting platform up to the WG leads, assuming we can get meetings recorded.

Would this mean whoever is WG lead just uses their personal link for whatever meeting platform they want to use in the meeting invite? What about groups where there is more than 1 WG lead, how do folks decide which link to provide?

re the conversation here https://docs.google.com/document/d/1Y7rug0XshcQPdKzptdWbQLQjcjgpFdLeEgP1nfkDAe4/edit?disco=AAAAG-_Jx4c - can we document the options somewhere here to make it less confusing if a meeting gets accidentally deleted again? @grantr mentioned to open an issue, but since the GSuite issues you already have are related I thought I'd check if this fits into that first before I do. Thanks!

**Comment by tcnghia** [comment]
*Update 12/2/2020*:
- I found no way to copy docs with all comments preserved. However, I could copy with new comments having the original author name and timestamp prepended (see https://github.com/knative/community/pull/346/files )

I believe that may be good enough historical contexts for us. If that is, we can start doing so. After copying and verifying the copies, we can start locking down the old drive and ask people to use the new drive entirely.

**Comment by vaikas** [comment]
Just to update the issue as well as jot down various slack conversations.
It was deemed necessary to have knative.team account logged in for us to be able to record meetings. So, each of the WG leads has been created a *.knative.team email account.
The one remaining issue is that for the meetings to be recordable, they seem to have to be owned / created by *knative.team account.

The task list here should be revisited and updated, and for ease of access, perhaps add an update to the end of the issue to see the current state of tasks that are still necessary to complete to call this item good?
https://github.com/knative/community/issues/86#issuecomment-649820387

**Comment by bsnchan** [comment]
@evankanderson Have you had a chance to start documenting the set up for the knative.team gsuite? If not, I'm happy to start one so we can close out this issue.

**Comment by evankanderson** [comment]
Feel free to start a doc or a PR; I'm happy to contribute both "what we do at the moment" and "historical things we did and why we moved away".

**Comment by bsnchan** [comment]
Hey @evankanderson - created a doc [here](https://docs.google.com/document/d/1N1jNn_9ILi85VsvlnmDhfyPjLfAK5VZAPKhSwTzs4-4/edit#) where I covered some content around how the Knative.team GSuite account is set up. Feel free to add any notes about the GDrive/GCal migration if you think it may be helpful for us in the future!

**Comment by bsnchan** [comment]
@evankanderson @tcnghia Hi you two - do you mind taking a look to see if there's any information that should be added re: migration?

**Comment by bsnchan** [comment]
This is pretty much done. @evankanderson suggested we convert this to markdown and PR this into the community repo so that the content is discoverable.
